### PR TITLE
add dust_limit config options

### PR DESCRIPTION
**Pull request by arowser**
Add support for set dust_limit from command or cnfigure file.

And its a workround for lnd -> lightning open channel fail when funding amount <57201 sat, see issue #2473

### CODE REVIEW

File: lightningd/options.c [hunk 1]
```diff
@@ -528,6 +548,8 @@ static const struct config testnet_config = {
 	.max_fee_multiplier = 10,
 
 	.use_dns = true,
+
+	.dust_limit = { 546 },
```
Comment (m-schmoock) [inline lightningd/options.c]: Just noticed, you missed to add the `dust_limit`  value to the wallet teststub config in `wallet/test/test_utils.c`.
It compiles without it, but Valgrind will be unhappy when a DB migration will be using this field in the future...

Comment (m-schmoock) [reply lightningd/options.c]: @niftynei    See? It will be missed   ;)

**Comment by arowser** [comment]
@m-schmoock  add dust_limit to test_utils.c, missed that, I check out from a old master.
I can't find a easy way to add test dust_limit config now, I'll add a testcase to lightning-integration after this pr merged,  and the issue must triger from lnd https://github.com/arowser/lightning-integration/tree/dust_limit

And lnd keep the value to 573 for support litecoin,  the developer known that, see https://github.com/lightningnetwork/lnd/issues/1142

**Comment by cdecker** [comment]
I don't quite get why the dust_limit needs to be configurable. Shouldn't this just be a simple change in `chainparams.c` to reflect the dust limit that is proper for that chain? Dust limits don't change all that often, and giving users more rope to hang themselves seems the wrong thing to do here.

**Comment by arowser** [comment]
Its make sense, I think this should be fixed in lnd side, its should set
different dust_limit for litecoin and bitcoin.

**Pull request closed without merging**